### DISCUSSION TITLE
Rpi pico-w socket.sendto() command fails every other powerup/reset

### DISCUSSION THREAD

**Discussion by hcgrant** [micropython/micropython#9974] [Hardware & Peripherals]
This code :--
```py
try :
        red_led.on() # diagnostic
        print("....Trying")
        skt.sendto(bytearray(message,encoding),(collector_IP, collector_port)) # send to collector
        print("done! ...")
        red_led.off() # diagnostic
except Exception as err :
        print("Sendto fail with error :  "+str(err))
```
fails every other poweron/reset except when running from thonny. The error code is  [Errno 113] EHOSTUNREACH
Creating the wifi link and the socket completes successfully.
I'm using a recent .uf2 file in a pico-w .... rp2-pico-w-20221111-unstable-v1.19.1-660-gc8913fdbf.uf2
Any ideas?

**Comment by karfas** [comment]
> fails every other poweron/reset except when running from thonny. The error code is [Errno 113] EHOSTUNREACH.

Maybe you should use another tool than thonny to analyze this issue.
I would also print all parameters of `sendto()` - just to be sure that they have the expected values. 

Once again, I  quote @peterhinch:
> If I can offer another general comment. This isn't intended as a criticism of Thonny or any other IDE, but an IDE is another layer of software which has the potential for problems. 
> When confronted with a puzzling problem I recommend using the simplest, most reliable, tools available to access the device under test. In my opinion these are official mpremote and unofficial rshell; in extremis I use a terminal emulator - my choice is miniterm.py.

**Reply by hcgrant** [reply]
Hi karfas,  This is the code to set up the socket .. and the status output.
```
def create_UDP_socket() :     # Setup the socket
    s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM) # Create a UDP socket
    s.settimeout(5)           # need to keep retrying for time data
    return s
. . . 
skt=create_UDP_socket()
tprint("Socket info "+str(skt))
```
And the tprint output is ..
```
Socket info <socket state=5 timeout=5000 incoming=0 off=0>
```
This is the same output even when skt.sendto() is giving an error.

The code for tprint() is..
```def tprint(s):  # Prints to a log file and to stdout
    print(s)
    f=open(log_file,"at")
    f.write(s+"\n")
    f.close()
```
When powered only by a power pack connected to VSYS via a schottski diode (USB cable disconnected) , this log file saves diagnostic info.

Its worth pointing out that the error doesn't happen when connected to thonny, it only occurs when running from a power pack or powered from the USB cable and thonny is not running.

**Reply by karfas** [reply]
I have no experience at all with Pico's.

However, This leaves the possibilities
- Thonny does _something_ on the board 
- there is some problem with your power supply
  - when USB is connected, but without thonny running/connected
  - when running from the power pack.

You can most likely rule out the first point when using a dumb terminal program.

For the power things, I would try to *measure* the voltage on the processor. You can also try to add a large (~100-500uF) to the board's Vcc, so power spikes caused by e.g. network activity get buffered.
